server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRunnable.java

```diff
@@ -128,6 +133,15 @@ private Mono<Void> processMail(MailQueue.MailQueueItem queueItem) {
                     // DO NOT CHANGE THIS to catch Error!
                     // For example, if there were an OutOfMemory condition caused because
                     // something else in the server was abusing memory, we would not want to start purging the retrying spool!
+                    AuditTrail.entry()
+                        .protocol("mailetcontainer")
+                        .action("RemoteDelivery")
+                        .parameters(ImmutableMap.of("mailId", mail.getName(),
+                            "mimeMessageId", mail.getMessage().getMessageID(),
+                            "sender", mail.getMaybeSender().asString(),
+                            "recipients", StringUtils.join(mail.getRecipients())))
```

Comment:
```suggestion
                        .parameters(() -> ImmutableMap.of("mailId", mail.getName(),
                            "mimeMessageId", mail.getMessage().getMessageID(),
                            "sender", mail.getMaybeSender().asString(),
                            "recipients", StringUtils.join(mail.getRecipients())))
```

Otherwize you would always join strings?